### PR TITLE
TOPPView: new data tab for DIA/OSW data

### PR DESCRIPTION
**Pull request by cbielow**
# Description

The new tab can currently show the protein/peptide/feature/transition tree
(not yet reacting to clicks, i.e. showing the chromatograms -- to come)

Also renames some signal/slots to more meaningful names, i.e.
`showSpectrumAs1D` was used for single spectra AND (wrongly) for multiple chromatograms, and it always meant opening a new Window/PlottingWidget.
Thus renamed to showSpectrumAsNew1D and showChromsAsNew1D.

Some other minor refactorings to increase shared code.

### CODE REVIEW

File: src/openms_gui/source/VISUAL/DIATreeTab.cpp [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+    {
+      for (int prot_index = 0; prot_index < data->getProteins().size(); ++prot_index)
+      {
+        const auto& prot = data->getProteins()[prot_index];
+        auto item_prot = createProt(prot, prot_index);
+        dia_treewidget_->addTopLevelItem(item_prot);
+      }
+    }
+
+
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/DIATreeTab.cpp:357]: there sare some unndeded multiple newlines here and after populateSearchBox

File: src/openms_gui/source/VISUAL/DIATreeTab.cpp [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+    spectra_combo_box_->setCurrentIndex(current_index);
+  }
+
+  void DIATreeTab::clear()
+  {
+    dia_treewidget_->clear();
+    spectra_combo_box_->clear();
+  }
+
+
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/DIATreeTab.cpp:381]: empty lines

File: src/openms_gui/source/VISUAL/TVDIATreeTabController.cpp [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+    // retrieve chromatogram (either from in-memory or on-disc representation)
+    MSChromatogram current_chrom;
+    current_chrom = exp_sptr->getChromatograms()[index];
+    if (current_chrom.empty() )
+    {
+      current_chrom = ondisc_sptr->getChromatogram(index);
+    }
+
+    // fill "dummy" spectrum with chromatogram data
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/TVDIATreeTabController.cpp:72]: this hack is so annoying ;) hope we get rid of it at some point

Comment (cbielow) [reply src/openms_gui/source/VISUAL/TVDIATreeTabController.cpp:72]: :-)

**Review by timosachsenberg — APPROVED** [review]

File: src/openms_gui/source/VISUAL/DIATreeTab.cpp [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+    enum HeaderNames
+    { // indices into QTableWidget's columns (which start at index 0)
+      ENTITY, INDEX, CHARGE, FULL_NAME, RT_DELTA, QVALUE, /* last entry --> */ SIZE_OF_HEADERNAMES
+    };
+    // keep in SYNC with enum HeaderNames
+    const QStringList HEADER_NAMES = QStringList()
+      << "entity" << "index" << "charge" << "full name" << "rt delta" << "q-value";
+  }
+
+  namespace Entity
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/DIATreeTab.cpp:59]: maybe document this in the next PR

**Comment by timosachsenberg** [comment]
http://cdash.openms.de/viewBuildError.php?buildid=71535

**Pull request merged**